dev/ssl_darwin_test.go

```diff
@@ -0,0 +1,76 @@
+package dev
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	. "github.com/puma/puma-dev/dev/devtest"
+	"github.com/puma/puma-dev/homedir"
+	"github.com/stretchr/testify/assert"
+)
+
+func deleteAllPumaDevCAFromDefaultKeychain() {
+	forEachPumaDevKeychainCertSha := func(subCommand string) string {
+		return fmt.Sprintf(`for sha in $(security find-certificate -a -c "Puma-dev CA" -Z | awk '/SHA-1/ {print $3}'); do %s; done`, subCommand)
+	}
+
+	if err := exec.Command("sh", "-c", forEachPumaDevKeychainCertSha("security delete-certificate -Z $sha")).Run(); err != nil {
+		panic(err)
+	}
+
+	log.Println("! NOTICE - REMOVED ALL CERTS LIKE \"Puma-dev CA\" FROM THE DEFAULT macOS KEYCHAIN")
+}
+
+func TestSetupOurCert_ensureNotWorldReadable(t *testing.T) {
+	t.Skip("not implemented yet - https://github.com/puma/puma-dev/issues/215")
```

Comment:
I'm not sure _exactly_ what the attack chain looks like, but if this CA cert and key remain unprivileged user-readable then it stands to reason that a piece of malicious software could read it, sign a new cert for www.whereyoukeepyoursecrets.com with it, hijack your DNS, and you'd be in the 💩right quick and likely never know it.